File: C/srcs/Sully.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>
#ifndef X
#define X x
#endif

int main()
{
	int		x = 5;
	char	*name;
	char	*gcc;
	char	*exec;
	char data[] = "#include <stdio.h>%1$c#include <string.h>%1$c#include <stdlib.h>%1$c#include <fcntl.h>%1$c#include <unistd.h>%1$c#ifndef X%1$c#define X x%1$c#endif%1$c%1$cint main()%1$c{%1$c	int		x = %5$d;%1$c	char	*name;%1$c	char	*gcc;%1$c	char	*exec;%1$c	char data[] = %2$c%4$s%2$c;%1$c	if (x > 0)%1$c	{%1$c		asprintf(&name, %2$cSully_%3$cd.c%2$c, X);%1$c		int	fd = open(name, O_CREAT | O_WRONLY | O_TRUNC, 0666);%1$c		free(name);%1$c		if (fd >= 0)%1$c		{%1$c			dprintf(fd, data, 10, 34, 37, data, X);%1$c			close(fd);%1$c			asprintf(&gcc, %2$cgcc -DX=x-1 -o Sully_%3$c1$d Sully_%3$c1$d.c%2$c, X);%1$c			system(gcc);%1$c			asprintf(&exec, %2$c./Sully_%3$cd%2$c, X);%1$c			system(exec);%1$c			free(gcc);%1$c			free(exec);%1$c		}%1$c	}%1$c}%1$c";
	if (x > 0)
	{
		asprintf(&name, "Sully_%d.c", X);
		int	fd = open(name, O_CREAT | O_WRONLY | O_TRUNC, 0666);
		free(name);
		if (fd >= 0)
		{
			dprintf(fd, data, 10, 34, 37, data, X);
			close(fd);
			asprintf(&gcc, "gcc -DX=x-1 -o Sully_%1$d Sully_%1$d.c", X);
			system(gcc);
			asprintf(&exec, "./Sully_%d", X);
			system(exec);
			free(gcc);
			free(exec);
		}
	}
}

File: Python/Sully.py
import os

def main():
    i = 5
    if __debug__ == False:
        i = i - 1
    s = "import os{0:c}{0:c}def main():{0:c}    i = {1}{0:c}    if __debug__ == False:{0:c}        i = i - 1{0:c}    s = {2:c}{3:s}{2:c}{0:c}    if i >= 0:{0:c}        fd = open('Sully_' + str(i) + '.py', 'w+'){0:c}        print(s.format(0x0a, i, 0x22, s), file=fd){0:c}        fd.close(){0:c}        os.system('python3 -O Sully_' + str(i) + '.py'){0:c}{0:c}if __name__ == '__main__':{0:c}	main()"
    if i >= 0:
        fd = open('Sully_' + str(i) + '.py', 'w+')
        print(s.format(0x0a, i, 0x22, s), file=fd)
        fd.close()
        os.system('python3 -O Sully_' + str(i) + '.py')

if __name__ == '__main__':
	main()
